lib/ret_web/channels/hub_channel.ex

```diff
@@ -93,9 +93,10 @@ defmodule RetWeb.HubChannel do
     {:reply, {:ok, %{has_remaining_subscriptions: has_remaining_subscriptions}}, socket}
   end
 
-  def handle_in("pin", %{"id" => object_id, "gltf_node" => gltf_node}, socket) do
+  def handle_in("pin", %{"id" => object_id, "gltf_node" => gltf_node, "token" => token}, socket) do
+    {:ok, %Account{} = account, _claims} = Guardian.resource_from_token(token)
     hub = socket |> hub_for_socket
-    RoomObject.perform_pin!(hub, %{object_id: object_id, gltf_node: gltf_node})
+    RoomObject.perform_pin!(hub, account, %{object_id: object_id, gltf_node: gltf_node})
```

Comment:
is there a reason you didn't want to create a separate `auth` message that associates the user with the socket? I think this is fine for now since we only have one authenticated message but I'm guessing that's what we'll want eventually

Reply:
Ah, Initially I bound the creds to the socket via `socketSettings` on the client side and then used `Guardian.Phoenix.Socket.authenticate(socket, ...)` on connect, but realized that I'd have to re-connect the socket if the user decided to sign out, which would obviously require them to re-join the room.
I hadn't considered a separate auth message and re-associating the socket. I'll try that.

Reply:
if you have a sign_in and sign_out message you can just update the `socket.assigns` to include the `account_id` I think

Reply:
and you'll need to also handle it in `join` as well, so rejoins will work -- you'll just need to have the token passed in optionally

Reply:
this is common enough i'd also just look at the phoenix examples out there -- i think there's a happy path here with the socket key being the user id. i am not sure if we want to go that route yet, since it will have an effect on presence.